File: backend/src/schemas/history.py
from typing import Optional

from pydantic import ConfigDict

from src.schemas.base_schema import StrictModel


class HistoryResponse(StrictModel):
    id_url: str
    year: int
    language: str
    title: Optional[str] = None
    content: str

    model_config = ConfigDict(from_attributes=True)


class HistoryCreate(StrictModel):
    year: int
    language: str
    title: Optional[str] = None
    content: str


class HistoryUpdate(StrictModel):
    year: Optional[int] = None
    language: Optional[str] = None
    title: Optional[str] = None
    content: Optional[str] = None

File: backend/src/schemas/history.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Clone)]
pub struct HistoryResponse {
    pub year: i32,
    pub language: String,
    pub title: Option<String>,
    pub content: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct HistoryCreate {
    pub year: i32,
    pub language: String,
    pub title: Option<String>,
    pub content: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct HistoryListQuery {
    pub language: Option<String>,
}
